source/mesh_refinement/isoterms.cc

```diff
@@ -0,0 +1,343 @@
+                cell->clear_coarsen_flag ();
+              if (refine)
+                cell->set_refine_flag ();
+            }
+        }
+    }
+
+    template <int dim>
+    void
+    Isoterms<dim>::
+    declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Mesh refinement");
+      {
+
+        prm.enter_subsection("Isoterms");
+        {
+ 
+          prm.declare_entry ("Exclude composition", "-1",
+                             Patterns::Double (),
+                             "Number of mininum and maximum areas defined by isoterms are going to be defined."
+                             );
+		  prm.declare_entry ("List of isoterms", "",
+                             Patterns::List(Patterns::List(Patterns::Anything(),0,100000000,","),0,100000000,";"),
+                             "Number of mininum and maximum areas defined by isoterms are going to be defined."
+                             );
+
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+    template <int dim>
+    void
+    Isoterms<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Mesh refinement");
+      {
+	minimum_refinement_level =Utilities::string_to_int(prm.get ("Minimum refinement level"));
+	maximum_refinement_level = Utilities::string_to_int(prm.get("Initial global refinement")) + Utilities::string_to_int(prm.get("Initial adaptive refinement"));
+        prm.enter_subsection("Isoterms");
+        {
+          i_excludeComposition = Utilities::string_to_int(prm.get("Exclude composition"));
+          n_isoterms_outer = Utilities::split_string_list(prm.get ("List of isoterms"),';');
+          // process the List of isoterms to get all data out of the structure
+	  int size_element = n_isoterms_outer.size();
+	  isoterms.resize(size_element,std::vector<std::string>(4,"0"));
+	  // loop through all given isoterms
+	  for ( unsigned int it1 = 0; it1 != n_isoterms_outer.size(); ++it1 )
+	  {
+	    n_isoterms_inner = Utilities::split_string_list(n_isoterms_outer[it1]);
+	    // The minimum and maximum refinemnt levels respecively
+	    isoterms[it1][0]=n_isoterms_inner[0];
+	    isoterms[it1][1]=n_isoterms_inner[1];
+	    // The temperatures for the min and max isoterms respectively
+	    isoterms[it1][2]=n_isoterms_inner[2];
+	    isoterms[it1][3]=n_isoterms_inner[3];
+
+	  }
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace MeshRefinement
+  {
+    ASPECT_REGISTER_MESH_REFINEMENT_CRITERION(Isoterms,
+                                              "isoterms",
+                                              "A mesh refinement criterion that ensures a "
+                                              "maximum and minimum refinement level between"
+                                              "two temperatures (isoterms), with the posibility"
```

Comment:
The compiler will concatenate all of these strings as you write them. In other words, you probably want a space at the end (or beginning, though our convention is the end) of each of these lines so that the output has spaces between words.